power_grid_model_c/power_grid_model/include/power_grid_model/common/enum.hpp

```diff
@@ -98,4 +98,16 @@ enum class CType : IntS { c_int32 = 0, c_int8 = 1, c_double = 2, c_double3 = 3 }
 
 enum class SerializationFormat : IntS { json = 0, msgpack = 1 };
 
+enum class OptimizerType : IntS {
+    no_optimization = 0,          // do nothing
+    automatic_tap_adjustment = 1, // power flow with automatic tap adjustment
+};
+
+enum class OptimizerStrategy : IntS {
+    any = 0,
+    lowest = 1,
+    highest = 2,
+    nan = na_IntS,
+};
```

Comment:
the problem here is that this will become part of the public API. IMO, this is similar to how `newton_raphson` calclation method can be used for both power flow and state estimation, but not for short circuit (as of yet). They are indeed specific, but not specific enough to restrict to only this optimizer type

Reply:
`minimum`, `maximum`

Reply:
any -> `any_optimum`